src/DatabaseDriver.php

```diff
@@ -1702,27 +1702,7 @@ protected function quoteNameString($name, $asSinglePart = false)
 	 */
 	protected function quoteNameStr($strArr)
 	{
-		$parts = [];
-		$q     = $this->nameQuote;
-
-		foreach ($strArr as $part)
-		{
-			if ($part === null)
-			{
-				continue;
-			}
-
-			if (\strlen($q) === 1)
-			{
-				$parts[] = $q . $part . $q;
-			}
-			else
-			{
-				$parts[] = $q[0] . $part . $q[1];
-			}
-		}
-
-		return implode('.', $parts);
+		return $this->quoteNameString(implode('.', $strArr));
```

Comment:
```suggestion
		$parts = [];

		foreach ($strArr as $part) {
			if ($part === null) {
				continue;
			}

			$parts[] = quoteNameString($part, true);
		}

		return implode('.', $parts);
```
This would be fully b/c to the old deprecated method, also for dots in the string or for null.